### PR TITLE
edge node auto register

### LINKED ISSUE #881: Create and register node from edge part
**What would you like to be added**:
Now when install kubeedge, after install the cloud part, if user want to add a edge node, must create node object by user self. 
Consider create and register node all by edge part, like kubelet done in native k8s.

**Why is this needed**:
User only install edge part, no need create node manually.

### PR DESCRIPTION
**Pull request by kuramal**
**What type of PR is this?**

/kind documentation
/kind feature

**What this PR does / why we need it**:
support edge node auto register

**Which issue(s) this PR fixes**:
Fixes #881

**Does this PR introduce a user-facing change?**:

```release-note
Edge node auto register
Now the edge node will automatically register by default,  If you want register manually, you should set the config register-node of edged with false
```

### CODE REVIEW

File: docs/setup/setup.md [hunk 1]
```diff
@@ -160,44 +160,10 @@ The cert/key will be generated in the `/etc/kubeedge/ca` and `/etc/kubeedge/cert
...
-      }
-    }
-    ```
-    **Note: you need to remember `metadata.name` , because edgecore need it**.
-+ If role is not set for the node, the pods, configmaps and secrets created/updated in the cloud cannot be synced with the node they are targeted for.
-+ Deploy edge node (**you must run on cloud side**)
-
-    ```shell
-    kubectl apply -f ~/cmd/yaml/node.json
-    ```
```
Comment (fisherxu) [inline docs/setup/setup.md]: We can make manually creating node as an option, and add a flag `register-node` to edgecore/edged. @kuramal @kadisi

Comment (kuramal) [reply docs/setup/setup.md]: Is there any situation where the user must register manually？ @fisherxu

Comment (fisherxu) [reply docs/setup/setup.md]: Yes, in some scenarios users need to manually create a node, they need to set some configuration in node object. Xrf: kubelet also has a flag to set it. https://github.com/kubernetes/kubernetes/blob/master/cmd/kubelet/app/options/options.go#L382

Comment (kadisi) [reply docs/setup/setup.md]: sounds good

File: cloud/pkg/edgecontroller/controller/upstream.go [hunk 2]
```diff
@@ -349,7 +356,44 @@ func (uc *UpstreamController) updateNodeStatus() {
 			}
 
 			switch msg.GetOperation() {
+			case model.InsertOperation:
+				_, err := uc.kubeClient.CoreV1().Nodes().Get(name, metaV1.GetOptions{})
+				if err == nil {
+					klog.Infof("node: %s already existed, do nothing", name)
```
Comment (kevin-wangzefeng) [inline cloud/pkg/edgecontroller/controller/upstream.go]:
```suggestion
					klog.Infof("node: %s already exists, do nothing", name)
```

**Comment by kevin-wangzefeng** [comment]
Please add release-note description

File: edge/conf/edge.yaml [hunk 3]
```diff
@@ -33,6 +33,7 @@ edged:
     hostname-override: edge-node
     interface-name: eth0
     edged-memory-capacity-bytes: 7852396000
+    register-node: true
```
Comment (kadisi) [inline edge/conf/edge.yaml]: now we are doing kubeedge component new api config feature, i suggest to add `register-node` field to new api struct.
https://github.com/kubeedge/kubeedge/blob/b61e49266715e493bbb5b2dc7593cc8c8289b8cc/pkg/apis/edgecore/v1alpha1/types.go#L103

Comment (kadisi) [reply edge/conf/edge.yaml]: if  this `register-node` field is not configured,  is it automatically registered by default? I suggest to refer to kubelet。

Comment (fisherxu) [reply edge/conf/edge.yaml]: > i suggest to add register-node field to new api struct.

Agreed.

> if this register-node field is not configured, is it automatically registered by default? I suggest to refer to kubelet

Let's set the `register-node` default as `true`.

File: cloud/pkg/edgecontroller/controller/upstream.go (hunk 2 above)
Comment (kuramal) [reply cloud/pkg/edgecontroller/controller/upstream.go]: @kevin-wangzefeng , I didn't deal with this suggested change in time before a force-push, so    I had to change it manually.

File: docs/setup/setup.md (hunk 1 above)
Comment (kuramal) [reply docs/setup/setup.md]: @fisherxu ,  I changed the node auto register to optional, PTAL

File: edge/conf/edge.yaml (hunk 3 above)
Comment (kuramal) [reply edge/conf/edge.yaml]: done, PTAL

**Comment by kuramal** [comment]
@fisherxu @kadisi update with new config api, PTAL

File: edgesite/conf/edgeSite.yaml [hunk 4]
```diff
@@ -25,6 +25,7 @@ edged:
     register-node-namespace: default
     hostname-override: edge-node
     interface-name: eth0
+    register-node: false
```
Comment (fisherxu) [inline edgesite/conf/edgeSite.yaml]: Does this change still needed?

File: edge/conf/edge.yaml (hunk 3 above)
Comment (fisherxu) [inline edge/conf/edge.yaml]: Does this change still needed?

File: docs/setup/deploy-edge-node.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
+# Deploy Edge Node Manually
```
Comment (fisherxu) [inline docs/setup/deploy-edge-node.md]: How to jump here from setup.md? I'd suggest to keep it in setup.md. For example:

#### Deploy the edge node
Edge node can be registered automatically.
##### Deploy Edge Node Manually
If you want to deploy edge node manually, here is an example.

Comment (fisherxu) [reply docs/setup/deploy-edge-node.md]: please help modify it more clear，like: L5-L16, we can just delete it.

File: edgesite/conf/edgeSite.yaml (hunk 4 above)
Comment (fisherxu) [reply edgesite/conf/edgeSite.yaml]: we can delete it in another PR, omit this comment.

File: edge/conf/edge.yaml (hunk 3 above)
Comment (fisherxu) [reply edge/conf/edge.yaml]: omit this comment.

**Pull request merged**